regular_expressions_basics/regular_expressions_basics.md

```diff
@@ -0,0 +1,338 @@
+***
+<div class = "answer">
+
+Let's look at the regular expression.
+
+* `A-Za-z` indicates that all unaccented upper and lowercase Latin letters are included, so `o` and `O` are valid for the first part of an email address.  However, `ö` is left out and not included.
+* `\d` indicates that any digit is acceptable as well, so `9` is valid.
+* `\-` indicates that a literal hyphen / dash is included in our character set, so `-` is valid.
+* `\.` indicates that a literal period is included in our character set, so `.` is valid.
+* `_` indicates an underscore, so `_` is also acceptable.
+
+There is no mention of a literal backslash, exclamation point, or space, so those are all excluded, as is our letter with an umlaut: `ö`.
+
+</div>
+***
+
+### Quantifiers: How Many Characters?
+
+So far, we've gone over single characters like `7`, metasequences like `\d`, and ranges like `[0-9]`, all to define what's valid for a single character in a pattern.  These ways to define what a character could be are collectively called **tokens**.  
+
+However, there's more to patterns than defining tokens one character at a time.  Usually, patterns have a certain number of characters in a row that are all represented by the same token, like five digits or three letters.  It would be very annoying to have to put `\d\d\d\d\d` (five digits) or `[A-Za-z][A-Za-z][A-Za-z]` (three letters) to represent these patterns.  And what if there's a varying length?  Three to five digits, or four or more letters?  That's where **quantifiers** come in!
+
+Sometimes we know there should be exactly one of something (for example, exactly one `@` in an email address).  Sometimes there can be an optional character (like the optional parentheses around the area code of a U.S. phone number).  Sometimes there are a specific number of characters from the same group (like exactly three letters or five digits).  Maybe there's a range, such as at least 2 letters but not more than 20.  
+
+Regular expressions allow you to quantify characters in a pattern.  You do this by adding a quantifier after your token.
+
+* An optional token, which might not appear at all or might appear at most once can be represented with a `?` quantifier.  For example, a phone number pattern that might begin with the number 1 or might omit it could begin with `1?`
+* An optional token that could repeat zero, one, or multiple times can be quantified with an asterisk (`*`).  For example, a phone number can have an optional extension that has one, two, or more digits.  That could be represented as `\d*`.
+* At least one character, maybe more? That's `+`. So, a first name has to have at least one letter, but could have more: `[A-Za-z]+`
+* A specific range, like one to three, but not less than one or more than three? That's `{1,3}`.  For example, maybe you know that in your health system, medical record numbers could have 6, 7, or 8 digits: `\d{6,8}`
```

Comment:
Since the quiz question uses a 2 digit number, maybe the example medical record number could have 9,10, or 11 digits so that learners can see that these quantifier numbers (unlike those in ranges) can be more than one character.

Reply:
I like it!  Great idea!